drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2528 @@
+		if (rc < 0) {
+			LOG_ERR("VBUS discharge pin configured to OUTPUT INACTIVE failed");
+			return rc;
+		}
+	}
+
+	/* Configure VCONN discharge pin to OUTPUT INACTIVE */
+	spec = &config->utcpd.gpios.vconn_discharge;
+	if (spec->port != NULL) {
+		if (!gpio_is_ready_dt(spec)) {
+			LOG_ERR("VCONN discharge pin port device not ready");
+			return -ENODEV;
+		}
+		rc = gpio_pin_configure_dt(spec, GPIO_OUTPUT_INACTIVE);
+		if (rc < 0) {
+			LOG_ERR("VCONN discharge pin configured to OUTPUT INACTIVE failed");
+			return rc;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Initializes UTCPD PHY
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_utcpd_phy_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t phyctl = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PHYCTL);
+
+	/* Enable PHY
+	 *
+	 * NOTE: Only UTCPD0 is supported.
+	 */
+	SYS->UTCPDCTL |= SYS_UTCPDCTL_POREN0_Msk;
+	phyctl |= UTCPD_PHYCTL_PHYPWR_Msk;
+	return NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, PHYCTL, phyctl);
+}
+
+/**
+ * @brief Checks if UTCPD Dead Battery mode is enabled
+ *
+ * @retval 1 Dead Battery mode is enabled
+ * @retval 0 Dead Battery mode is not enabled
+ */
+static int numaker_utcpd_deadbattery_query_enable(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t phyctl = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PHYCTL);
+
+	return (phyctl & UTCPD_PHYCTL_DBCTL_Msk) ? 0 : 1;
+}
+
+/**
+ * @brief Enables or disables UTCPD Dead Battery mode
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_utcpd_deadbattery_set_enable(const struct device *dev, bool enable)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t phyctl = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PHYCTL);
+
+	if (enable) {
+		/* Dead Battery circuit controls internal Rd/Rp */
+		phyctl &= ~UTCPD_PHYCTL_DBCTL_Msk;
+	} else {
+		/* UTCPD.ROLCTL controls internal Rd/Rp */
+		phyctl |= UTCPD_PHYCTL_DBCTL_Msk;
+	}
+	return NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, PHYCTL, phyctl);
+}
+
+/**
+ * @brief Initializes UTCPD Dead Battery mode
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_utcpd_deadbattery_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+
+	return numaker_utcpd_deadbattery_set_enable(dev, config->utcpd.dead_battery);
+}
+
+/**
+ * @brief Initializes UTCPD interrupts
+ *
+ * @retval 0 on success
+ * @retval -EIO on failure
+ */
+static int numaker_utcpd_interrupts_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	int rc;
+	uint32_t ie;
+	uint32_t pwrstsie;
+	uint32_t futstsie;
+	uint32_t vndie;
+
+	/* IE */
```

Comment:
these comments are not very useful, just drop them, make the code a bit more compact

Reply:
removed